src/components/SelectionList/Search/ReportListItem.tsx

```diff
@@ -74,14 +76,14 @@ function ReportListItem<TItem extends ListItem>({
     const openReportInRHP = (transactionItem: TransactionListItemType) => {
         const backTo = Navigation.getActiveRoute();
 
-        const isFromSelfDM = transactionItem.reportID === CONST.REPORT.UNREPORTED_REPORT_ID;
+        const reportID = getReportIDForTransaction(transactionItem);
+        const siblingTransactionThreadIDs = reportItem.transactions.map(getReportIDForTransaction);
 
-        const reportID =
-            (!transactionItem.isFromOneTransactionReport || isFromSelfDM) && transactionItem.transactionThreadReportID !== CONST.REPORT.UNREPORTED_REPORT_ID
-                ? transactionItem.transactionThreadReportID
-                : transactionItem.reportID;
-
-        Navigation.navigate(ROUTES.SEARCH_REPORT.getRoute({reportID, backTo}));
+        // When opening the transaction thread in RHP we need to find every other ID for the rest of transactions
+        // to display prev/next arrows in RHP for navigation
+        setActiveTransactionThreadIDs(siblingTransactionThreadIDs).then(() => {
```

Comment:
I wonder, could you test with a report that will have 300 expenses for example. How long will this processing take before we navigate to the RHP? @Kicu @allroundexperts

Reply:
Sure I can test. However I remember in the past asking about such cases in the context of other tasks, and the answer was that they won't happen.
Honestly I have no idea how the table view would behave if it got 300 transactions